Common/src/CConfig.cpp

```diff
@@ -3768,22 +3810,102 @@ void CConfig::SetPostprocessing(SU2_COMPONENT val_software, unsigned short val_i
+      default:
+        if (nSpecies_Init + 1 != 1) SU2_MPI::Error("Viscosity model not available.", CURRENT_FUNCTION);
+        break;
+    }
+
+    switch (Kind_ConductivityModel) {
+      case CONDUCTIVITYMODEL::CONSTANT:
+        if ((Kind_ConductivityModel_Turb == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) && (Kind_Turb_Model != TURB_MODEL::NONE)) {
+          if ((nThermal_Conductivity_Constant != nSpecies_Init + 1) || (nPrandtl_Turb != nSpecies_Init + 1)) {
+            SU2_MPI::Error(
+                "The use of FLUID_MIXTURE requires the number of entries for THERMAL_CONDUCTIVITY_CONSTANT and "
+                "PRANDTL_TURB,\n"
+                "to be equal to the number of entries of SPECIES_INIT + 1",
+                CURRENT_FUNCTION);
+          }
+        } else {
+          if (nThermal_Conductivity_Constant != nSpecies_Init + 1) {
+            SU2_MPI::Error(
+                "The use of FLUID_MIXTURE requires the number of entries for THERMAL_CONDUCTIVITY_CONSTANT,\n"
+                "to be equal to the number of entries of SPECIES_INIT + 1",
+                CURRENT_FUNCTION);
+          }
+        }
+        break;
+      case CONDUCTIVITYMODEL::CONSTANT_PRANDTL:
+        if (Kind_ConductivityModel_Turb == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          if ((nPrandtl_Lam != nSpecies_Init + 1) || (nPrandtl_Turb != nSpecies_Init + 1)) {
+            SU2_MPI::Error(
+                "The use of FLUID_MIXTURE requires the number of entries for PRANDTL_LAM and PRANDTL_TURB,\n"
+                "to be equal to the number of entries of SPECIES_INIT + 1",
+                CURRENT_FUNCTION);
+          }
+        } else {
+          if (nPrandtl_Lam != nSpecies_Init + 1) {
+            SU2_MPI::Error(
+                "The use of FLUID_MIXTURE requires the number of entries for PRANDTL_LAM,\n"
+                "to be equal to the number of entries of SPECIES_INIT + 1",
+                CURRENT_FUNCTION);
+          }
+        }
+        break;
+      default:
+        if (nSpecies_Init + 1 != 1) SU2_MPI::Error("Conductivity model not available.", CURRENT_FUNCTION);
+        break;
+    }
   }
 
   /*--- Overrule the default values for viscosity if the US measurement system is used. ---*/
 
   if (SystemMeasurements == US) {
-
-    /* Correct the viscosities, if they contain the default SI values. */
-    if(fabs(Mu_Constant-1.716E-5) < 1.0E-15) Mu_Constant /= 47.88025898;
-    if(fabs(Mu_Ref-1.716E-5)      < 1.0E-15) Mu_Ref      /= 47.88025898;
-
-    /* Correct the values with temperature dimension, if they contain the default SI values. */
-    if(fabs(Mu_Temperature_Ref-273.15) < 1.0E-8) Mu_Temperature_Ref *= 1.8;
-    if(fabs(Mu_S-110.4)                < 1.0E-8) Mu_S               *= 1.8;
-
-    /* Correct the thermal conductivity, if it contains the default SI value. */
-    if(fabs(Thermal_Conductivity_Constant-0.0257) < 1.0E-10) Thermal_Conductivity_Constant *= 0.577789317;
+    if (Kind_FluidModel == FLUID_MIXTURE) {
```

Comment:
Maybe I missed this, but where do the 47, 0.577 values come from?  It may be good to make this a variable (like Avogadros)...but Im not sure.

Reply:
Hi, These are the factors used for converting defaults values from SI system  to US system.